### PR TITLE
Leapp::Api::PreupgradeReportsController

### PR DESCRIPTION
**Pull request by stejskalleos**
Requires https://github.com/oamg/foreman_leapp/pull/17

Info:
- Remove obsole `create` action (it was used for `preupgrade callback`)
- Add `index`
- Add `rabl` templates
- Remove obsolete controllers

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-  namespace :api, :defaults => { :format => 'json' } do
-    scope '(:apiv)', :module => :v2, :defaults => { :apiv => 'v2' }, :apiv => /v2/,
-                     :constraints => ApiConstraints.new(:version => 2, :default => true) do
-      resources :preupgrade_reports, only: %i[show create]
+  namespace :api, defaults: { format: 'json' } do
+    scope '(:apiv)', module: :v2, defaults: { apiv: 'v2'}, apiv: /v1|v2/, constraints: ApiConstraints.new( version: 2, default: true) do
+      resources :preupgrade_reports, only: %i[index show]
+
       # XXX FIXME find a way to utilize built-in routing
+      # TODO: Do we actually need them?
```
Comment (xprazak2) [inline config/routes.rb]: I think these can be removed as well.

File: config/routes.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  namespace :api, :defaults => { :format => 'json' } do
-    scope '(:apiv)', :module => :v2, :defaults => { :apiv => 'v2' }, :apiv => /v2/,
-                     :constraints => ApiConstraints.new(:version => 2, :default => true) do
-      resources :preupgrade_reports, only: %i[show create]
+  namespace :api, defaults: { format: 'json' } do
+    scope '(:apiv)', module: :v2, defaults: { apiv: 'v2'}, apiv: /v1|v2/, constraints: ApiConstraints.new( version: 2, default: true) do
```
Comment (xprazak2) [inline config/routes.rb]: I think v1 was already dropped, having just v2 should be ok.

File: app/controllers/api/v2/preupgrade_reports_controller.rb [hunk 3]
```diff
@@ -4,35 +4,36 @@ module Api
...
+        @preupgrade_reports = resource_scope_for_index
       end
 
+      api :GET, '/preupgrade_reports/:id', N_('Show Preupgrade report')
+      param :id, :identifier, required: true
       def show
-        render :json => full_report(PreupgradeReport.find(params['id']))
       end
 
+      # TODO: Do we need it?
```
Comment (xprazak2) [inline app/controllers/api/v2/preupgrade_reports_controller.rb]: I do not think we need it.

**Comment by xprazak2** [comment]
Nice work, it would be nice to have some basic tests.

**Comment by stejskalleos** [comment]
Updated + added tests for Api controller.

**Comment by adamruzicka** [comment]
Thank you @stejskalleos & @xprazak2 !

**Pull request merged**